lib/faraday/adapter/test.rb

```diff
@@ -108,12 +108,20 @@ def matches?(request_uri, request_headers, request_body)
           request_params = request_query ?
             Faraday::Utils.parse_nested_query(request_query) :
             {}
-          request_path == path &&
+          path_match?(request_path) &&
             params_match?(request_params) &&
             (body.to_s.size.zero? || request_body == body) &&
             headers_match?(request_headers)
         end
 
+        def path_match?(request_path)
+          if path.is_a? Regexp
+            path =~ request_path
```

Comment:
Could we capture the MatchData object by doing `match = request_path.match(path)` and somehow pass it to the block? That way the block could access the values from capture groups